crates/swift-bridge-ir/src/bridged_type.rs

```diff
@@ -54,6 +54,13 @@ pub(crate) trait BridgeableType: Debug {
     /// Whether or not this is a `Result<T,E>`.
     fn is_result(&self) -> bool;
 
+    /// Extract the Swift representation of each variant from Rust's Result<T,E>.
```

Comment:
This exact Swift results variant feels a bit too specific to belong in the `BridgeableType` trait.

---

Here's what I'm thinking instead:

---

We add an `as_result` method to the `BridgeableType` trait

```rust
trait BridgeableType {
    // ...

    fn is_result(&self) -> bool {
        self.as_result().is_some()
    }

    fn as_result(&self) -> Option<&BuiltInResult>;
}
```

---

Then we can use `.as_result()` to get back the BuiltInResult, and use a method(s) on the built in result to generate whatever information we need when generating the async support code on the Swift side.